SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialWorkerFlags/SpatialWorkerFlagsTest.cpp

```diff
@@ -51,39 +51,160 @@ SPATIALWORKERFLAGS_TEST(GIVEN_a_flagUpdate_op_WHEN_removing_a_worker_flag_THEN_f
 	return true;
 }
 
-SPATIALWORKERFLAGS_TEST(GIVEN_a_bound_delegate_WHEN_a_worker_flag_updates_THEN_bound_function_invoked)
+// delegates
+
+// any flag
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_registered_any_flag_update_delegate_WHEN_any_worker_flag_updates_THEN_delegate_invoked)
 {
 	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
-	FOnWorkerFlagsUpdatedBP WorkerFlagDelegate;
-	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetFlagUpdated);
+	FOnAnyWorkerFlagUpdatedBP WorkerFlagDelegate;
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
 
 	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
-	SpatialWorkerFlags->BindToOnWorkerFlagsUpdated(WorkerFlagDelegate);
+	SpatialWorkerFlags->RegisterAnyFlagUpdatedCallback(WorkerFlagDelegate);
+
+	// Add test flag
+	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	// Add test flag 2
+	Worker_FlagUpdateOp OpAddFlag2 = CreateWorkerFlagUpdateOp("test2", "20");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag2);
+
+	TestTrue("Delegate Function was called twice", SpyObj->GetTimesFlagUpdated() == 2);
+
+	return true;
+}
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_registered_any_flag_update_delegate_WHEN_registering_the_delegate_THEN_delegate_is_not_invoked)
```

Comment:
WHEN_registering_the_delegate -> WHEN_delegate_is_unregistered 

If understand the test correctly I think this should change?

Reply:
you are right should be
`GIVEN_a_registered_any_flag_update_delegate_WHEN_unregistering_the_delegate_THEN_delegate_is_not_invoked`